metricflow/model/semantics/semantic_containers.py

```diff
@@ -67,59 +68,65 @@ def __init__(  # noqa: D
 
     def element_specs_for_metrics(
         self,
-        metric_specs: List[MetricSpec],
+        metric_references: List[MetricReference],
         with_any_property: FrozenSet[LinkableElementProperties] = LinkableElementProperties.all_properties(),
         without_any_property: FrozenSet[LinkableElementProperties] = frozenset(),
     ) -> List[LinkableInstanceSpec]:
         """Dimensions common to all metrics requested (intersection)"""
 
         all_linkable_specs = self._linkable_spec_resolver.get_linkable_elements_for_metrics(
-            metric_specs=metric_specs,
+            metric_references=metric_references,
             with_any_of=with_any_property,
             without_any_of=without_any_property,
         ).as_spec_set
 
         return sorted(all_linkable_specs.as_tuple, key=lambda x: x.qualified_name)
 
-    def get_metrics(self, metric_names: List[MetricSpec]) -> List[Metric]:  # noqa: D
+    def get_metrics(self, metric_references: List[MetricReference]) -> List[Metric]:  # noqa: D
         res = []
-        for metric_name in metric_names:
-            if metric_name not in self._metrics:
-                raise MetricNotFoundError(f"Unable to find metric `{metric_name}`. Perhaps it has not been registered")
-            res.append(self._metrics[metric_name])
+        for metric_reference in metric_references:
+            if metric_reference not in self._metrics:
+                raise MetricNotFoundError(
+                    f"Unable to find metric `{metric_reference}`. Perhaps it has not been registered"
+                )
+            res.append(self._metrics[metric_reference])
 
         return res
 
     @property
-    def metric_names(self) -> List[MetricSpec]:  # noqa: D
+    def metric_references(self) -> List[MetricReference]:  # noqa: D
         return list(self._metrics.keys())
 
-    def get_metric(self, metric_name: MetricSpec) -> Metric:  # noqa:D
-        if metric_name not in self._metrics:
-            raise MetricNotFoundError(f"Unable to find metric `{metric_name}`. Perhaps it has not been registered")
-        return self._metrics[metric_name]
+    def get_metric(self, metric_reference: MetricReference) -> Metric:  # noqa:D
+        if metric_reference not in self._metrics:
+            raise MetricNotFoundError(f"Unable to find metric `{metric_reference}`. Perhaps it has not been registered")
+        return self._metrics[metric_reference]
+
+    def get_metric_spec(self, metric_reference: MetricReference) -> MetricSpec:  # noqa: D
+        metric = self.get_metric(metric_reference)
+        return MetricSpec(element_name=metric.name)
```

Comment:
Will we be able to eliminate this in the future as well? Seems like an API we'd rather not have in the long run.